lib/backend.js

```diff
@@ -771,6 +771,17 @@ Backend.prototype.fetchSnapshot = function(agent, index, id, version, callback)
 Backend.prototype._fetchSnapshot = function(collection, id, version, callback) {
   var db = this.db;
   var backend = this;
+  var shouldGetLatestSnapshot = version === null;
+
+  if (shouldGetLatestSnapshot) {
+    var options = {metadata: true};
+    return backend.db.getSnapshot(collection, id, null, options, function(error, snapshot) {
+      if (error) return callback(error);
+
+      callback(error, snapshot);
```

Comment:
We've just checked, and we don't have an error, so let's explicitly pass `null`, otherwise it looks a little confusing (hooray...callbacks...)

```suggestion
      callback(null, snapshot);
```

Reply:
Changed